content/en/cosign/keyless.md

```diff
@@ -0,0 +1,177 @@
+4. Sign images in GCB, without keys!
+
+### Timestamps
+
+Signature timestamps are checked in the [rekor](https://github.com/sigstore/rekor) transparency log. Rekor's `IntegratedTime` is signed as part of its `signedEntryTimestamp`. Cosign verifies the signature over the timestamp and checks that the signature was created while the certificate was valid.
+
+## Upcoming work
+
+* Root CA hardening: We should use intermediate certs rather than the root, and support chained verification.
+* Root CA configuration: We should allow users to change the roots and add their own.
+* Other timestamps: We should allow for other timestamp attestations, including attached [RFC3161](https://www.ietf.org/rfc/rfc3161.txt) signatures.
+* Probably a lot more: This is very experimental.
+* More OIDC providers: Obvious.
+
+## Public Staging Environment
+
+
+There is a public staging environment that is running Fulcio, Rekor and OIDC issuer.
+
+**NOTE** The staging environment provides no SLO guarantees nor the same protection of the root key material for TUF. This environment is meant for development and testing only, PLEASE do not use for production purposes.
+
+The endpoints are as follows:
+
+* https://fulcio.sigstage.dev
+* https://rekor.sigstage.dev
+* https://oauth2.sigstage.dev/auth
+
+These instances are operated and maintained in the same manner as the public production environment for Sigstore.
+
+### Usage
+
+To use this instance, follow the steps below:
+
+1. `rm -r ~/.sigstore`
+1. `gsutil cp -r gs://tuf-root-staging/root.json .`
+1. `cd tuf-root-staging`
+1. `cosign initialize --mirror=tuf-root-staging --root=root.json`
+1. `COSIGN_EXPERIMENTAL=1 cosign sign --oidc-issuer "https://oauth2.sigstage.dev/auth" --fulcio-url "https://fulcio.sigstage.dev" --rekor-url "https://rekor.sigstage.dev" ${IMAGE_DIGEST}`
+1. `COSIGN_EXPERIMENTAL=1 cosign verify --rekor-url "https://rekor.sigstage.dev" ${IMAGE}`
+
+* Steps 1-4 configures your local environment to use the staging keys and certificates.
+* Step 5 specify the staging environment with flags needed for signing.
+* Step 6 specify the staging environment with flags needed for verifying.
+
+#### Revert back to Production
+
+We need to clear the local TUF root data and re-initialize with the default production TUF root data.
+
+1. `rm -r ~/.sigstore`
+1. `cosign initialize`
+
+## Custom Infrastructure
+
+If you're running your own sigstore services flags are available to set your own endpoint's, e.g
+
+```
+ COSIGN_EXPERIMENTAL=1 cosign sign -oidc-issuer "https://oauth2.example.com/auth" \
+                        -fulcio-url "https://fulcio.example.com" \
+                        -rekor-url "https://rekor.example.com"  \
+                        $IMAGE_DIGEST
+
+```
+
+### Custom root Cert
+
+You can override the public good instance root CA using the environment variable `SIGSTORE_ROOT_FILE`, e.g.
```

Comment:
We should update this section with the flags for the CT and Rekor keys too

Reply:
Will need to figure out what those flags are ... will update when committed and pushed.

Reply:
Have had a look at the current `cosign` that I've got locally from `brew` and I can't see any keys for rekor when doing `cosign sign -help`. Which keys are you referencing exactly? I can quickly put them in if possible